classes/Autocomplete.php

```diff
@@ -0,0 +1,164 @@
+        while ($row = $q->fetch_assoc()) {
+            if ($option == "new_var") {
+                $users[$key] = array('value' => $row[$subtype], 'label' => '', 'info' => '', 'group' => '');
+            } else {
+                if ($type == "instrument" || ($type == "variable" && $row["field_name"] != $row['form_name'] . "_complete" && $row["field_name"] != "record_id")) {
+                    // Trim all, just in case
+                    $label = trim(strtolower($row[$subtype]));
+                    $info = "";
+                    $group = "";
+                    if ($type == "variable") {
+                        $info = htmlspecialchars(trim($row['element_label']));
+                        $group = \REDCap::getInstrumentNames(trim(strtolower($row['form_name'])));
+                    } else {
+                        $label = \REDCap::getInstrumentNames(trim(strtolower($row[$subtype])));
+                    }
+                    // Calculate search match score.
+                    $userMatchScore[$key] = 0;
+
+                    // Loop through each search term for this person
+
+                    // Set length of this search string
+                    $this_term_len = strlen($this_term);
+                    // For partial matches, give +1 point for each letter
+                    if (strpos($row[$subtype], $this_term) !== false) {
+                        $userMatchScore[$key] = $userMatchScore[$key] + $this_term_len;
+                    }
+
+                    // Wrap any occurrence of search term in label with a tag
+                    $label = self::searchTerms($search_terms, $label);
+
+                    // Add to arrays
+                    $users[$key] = array(
+                        'value' => $row[$subtype],
+                        'label' => $label,
+                        'info' => $info,
+                        'group' => $group
+                    );
+                    $usernamesOnly[$key] = $row['username'];
+                    // If username, first name, or last name match EXACTLY, do a +100 on score.
+                    if (in_array($row[$subtype], $search_terms)) {
+                        $userMatchScore[$key] = $userMatchScore[$key] + 100;
+                    }
+
+                    // Increment key
+                    $key++;
+                }
+            }
+        }
+
+        // Sort users by score, then by username
+        if ($option == "old_var") {
+            $count_users = count($users);
+            if ($count_users > 0) {
+                // Sort
+                array_multisort($userMatchScore, SORT_NUMERIC, SORT_DESC, $usernamesOnly, SORT_STRING, $users);
+                // Limit only to X users to return
+                $limit_users = 10;
+                if ($count_users > $limit_users) {
+                    $users = array_slice($users, 0, $limit_users);
+                }
+            }
+        }
+        return json_encode($users);
```

Comment:
Since users are not returned, let's change these comments and variable names to be variables or instruments or some abstract term meaning both